backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/NonContainmentReferenceHandler.java

```diff
@@ -55,32 +60,48 @@ public NonContainmentReferenceHandler(MessageReporter messageReporter) {
     public boolean isNonContainmentReference(final Element owner, String referenceName, final JsonNode astValue) {
         if (astValue.isContainerNode()) {
             JsonNode referenceTypeNode = astValue.get("$type");
-            JsonNode textNode = astValue.get("text");
-            return referenceTypeNode != null && astValue.has("reference") && textNode != null && textNode.isTextual();
+            return referenceTypeNode != null && this.referenceNodeTester.test(astValue);
         }
         return false;
     }
 
     public void createProxy(final Element owner, String referenceName, final JsonNode astValue) {
-        JsonNode referenceTypeNode = astValue.get("$type");
-        JsonNode textNode = astValue.get("text");
-        String referenceType = referenceTypeNode.asText();
-        final EObject refrenceTypeInstance = this.typeBuilder.createObject(astValue);
-
-        if (refrenceTypeInstance instanceof InternalEObject internalTarget) {
-            String qualifiedNameTarget = textNode.asText();
-            internalTarget.eSetProxyURI(URI.createGenericURI("syson-import", QUALIFIED_CONST, qualifiedNameTarget));
-            // It should be a reference
-            Optional<EReference> optEReference = this.referenceComputer.getNonContainmentReference(owner.eClass(), refrenceTypeInstance.eClass(), referenceName);
-            if (optEReference.isEmpty()) {
-                this.messageReporter.error(MessageFormat.format("Unable to find a reference from {0} to {1} with reference name {2} on object {3}", owner.eClass().getName(),
-                        refrenceTypeInstance.eClass().getName(), referenceType, this.logNameProvider.getName(owner)));
+        if (this.isMetadataAccessExpressionCase(owner, referenceName, astValue)) {
```

Comment:
May be you could add a piece of code here that will handle this case and potential future similar cases?
Then move this code in the new class. The new class will clearly mention that it is workarounds for incompatibilities between SysIDE version and SysMLv2 metamodel version.